torch/_inductor/ir.py

```diff
@@ -5250,6 +5250,51 @@ def should_allocate(self):
         return False
 
 
+class TMADescriptor(ExternKernel):
+    def __init__(
+        self,
+        tensor: TensorBox,
+        dims: List[Union[int, torch.SymInt]],
+        block_dims: List[Union[int, torch.SymInt]],
+        element_size: Optional[Union[int, torch.SymInt]] = None,
+    ):
+        assert len(dims) in (1, 2)
+        assert len(dims) == len(block_dims)
+
+        if element_size is None:
+            element_size = tensor.get_dtype().itemsize
+
+        self.tensor = tensor
+        self.dims = dims
+        self.block_dims = block_dims
+        self.element_size = element_size
+        self.rank = len(self.dims)
+
+        inputs = [tensor]
+        constant_args = [
+            *self.dims,
+            *self.block_dims,
+            self.element_size,
+        ]
```

Comment:
Is dims here actually constant ?

Reply:
Not necessarily, they can be SymInts. Was following what the `UserDefinedTritonKernel` does, where all non-`TensorBox` args are put into the `constant_args` [here](https://github.com/pytorch/pytorch/blob/504904c9c66fc89eada912b1b7d9fdf1c3e56b14/torch/_inductor/ir.py#L5353).

I was unsure about the semantics of the `constant_args` parameter of `ExternKernel`. Looking into the code, seems `self.constant_args` is mostly used in the codegen-related methods, which are not relevant for `TMADescriptor` (neither for `UserDefinedTritonKernel`), as the codegen is overridden at the root. Although, I also see it being used as a potential source of unbacked SymInts [here](https://github.com/pytorch/pytorch/blob/504904c9c66fc89eada912b1b7d9fdf1c3e56b14/torch/_inductor/ir.py#L5131). So perhaps I should keep this code as is?